include/filewriter/summarywriter_impl.hpp

```diff
@@ -150,6 +150,129 @@ void SummaryWriter<Filewriter>::Image(const std::string& tag,
     remove(fileNames[i].c_str());
 }
 
+template<typename Filewriter>
+void SummaryWriter<Filewriter>::PrCurve(const std::string& tag,
+                                        const std::vector<double>& labels,
+                                        const std::vector<double>& predictions,
+                                        Filewriter& fw,
+                                        int threshold,
+                                        std::vector<double>weights,
+                                        const std::string& displayName,
+                                        const std::string& description)
+{
+    // Pr plugin.
+    mlboard::PrCurvePluginData *prCurvePlugin = new PrCurvePluginData();
+    prCurvePlugin->set_version(0);
+    prCurvePlugin->set_num_thresholds(threshold);
+    std::string pr_curve_content;
+    prCurvePlugin->SerializeToString(&pr_curve_content);
+
+    // PluginMeta data.
+    mlboard::SummaryMetadata_PluginData *pluginData =
+        new SummaryMetadata::PluginData();
+    pluginData->set_plugin_name("pr_curves");
+    pluginData->set_content(pr_curve_content);
+
+    // Summary Meta data.
+    mlboard::SummaryMetadata *metaData = new SummaryMetadata();
+    metaData->set_display_name(displayName == "" ? tag : displayName);
+    metaData->set_summary_description(description);
+    metaData->set_allocated_plugin_data(pluginData);
+
+    double minCount = 1e-7;
+    std::vector<std::vector<double>> data;
+    while (weights.size() < labels.size())
+    {
+        weights.push_back(1.0);
+    }
+    std::vector<double>edges;
+    mlboard::util::histogramEdges({0, (double)threshold - 1},
+      threshold, edges);
+    std::vector<double> tp(edges.size(), 0), fp(edges.size(), 0);
+
+    for (size_t i = 0; i < labels.size(); ++i)
+    {
+        float v = labels[i];
+        int item = predictions[i] * (threshold -1);
+        auto lb =
+            lower_bound(edges.begin(), edges.end(), item);
+        // Include the exact number in previous bucket.
+        if (*lb != item)
+            lb--;
+        tp[lb - edges.begin()] = tp[lb - edges.begin()] + (v * weights[i]);
+        fp[lb - edges.begin()] = fp[lb - edges.begin()] +
+            ((1 - v) * weights[i]);
+    }
+
+    // Reverse cummulative sum.
+    for (int i = tp.size() - 2; i >= 0; i--)
+    {
+        tp[i] = tp[i] + tp[i+1];
+        fp[i] = fp[i] + fp[i+1];
+    }
+    std::vector<double> tn(tp.size()), fn(tp.size()),
```

Comment:
```suggestion
    std::vector<double> tn(tp.size()), fn(tp.size()),
```

Same about it over here `trueNegative`, and `falseNegative` would be better I think. Let me know what you think.